File: playground.js
function makeListNode(length) {
    var root, parent;
    for (var i = 0; i < length; i++) {
        var node = new ListNode(i + 1);
        if (i == 0) {
            root = node;
            parent = root;
        }
        else {
            parent.next = node;
            parent = parent.next;
        }
    }
    return root;
}
function printListNode(node) {
    // console.log(node)
    var root, parent;
    root = node;
    parent = root;
    while (parent.next != undefined || parent.next != null) {
        console.log(parent.data);
        parent = parent.next;
    }
}
function main() {
    // console.log("Hello, Language TypeScript");
    var node = makeListNode(100);
    printListNode(node);
}
var ListNode = /** @class */ (function () {
    function ListNode(data) {
        this.data = data;
    }
    return ListNode;
}());
main();

File: playground.rs
fn main() {
    let root = make_list_node();
    root.print();
}

fn make_list_node() -> ListNode {
    let mut root: ListNode;
    let mut parent: ListNode;
    for i in 0..5 {
        let node = ListNode::from(i);
        if i == 0 {
            root = node;
            parent = node;
        } else {
            parent = node;
            parent.next = &mut parent;
        }
    }
    return root;
}

pub struct ListNode {
    pub data: i8,
    pub next: *const ListNode,
}

impl ListNode {
    pub fn print(&self) {
        let mut parent: *const ListNode = self;
        while (*parent).next != Option::None {
            println!("{}", (*parent).data);
            parent = parent.next;
        }
        println!("{}", (*parent).data);
    }
    pub fn from(data: i8) -> Self {
        return ListNode {
            data: data,
            next: Option::None,
        };
    }
}
